crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -63,6 +64,14 @@ pub fn execute_call_entry_point(
         Some(class_hash) => class_hash,
         None => storage_class_hash, // If not given, take the storage contract class hash.
     };
+
+    // region: Modified blockifier code
+    runtime_state
+        .cheatnet_state
+        .trace_data
+        .add_class_hash_to_current_call(class_hash);
+    // endregion
```

Comment:
The class hash being valid is already checked above in the code copied from the blockifier. It is the safest place to do it (mocked calls, calling non existing contracts etc.)